apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs

```diff
@@ -498,35 +523,26 @@ where
                     if !parent.schema.all_of.is_empty() {
                         tracing::warn!("No inheritance data found for `{}`", parent.schema.id);
                     }
-                    ontology_type_resolver.add_open(parent_id, Arc::new(parent.schema));
+                    ontology_type_resolver.add_unresolved(parent_id, Arc::new(parent.schema));
                 }
             });
 
-        for (data_type_id, inserted_data_type) in &inserted_data_types {
-            ontology_type_resolver.add_open(*data_type_id, Arc::clone(inserted_data_type));
-        }
-        let schema_metadata = inserted_data_types
+        let closed_schemas = inserted_data_types
             .iter()
-            .map(|(data_type_id, _)| {
-                ontology_type_resolver
+            .map(|(data_type_id, data_type)| {
+                let closed_metadata = ontology_type_resolver
                     .resolve_data_type_metadata(*data_type_id)
-                    .change_context(InsertionError)
+                    .change_context(InsertionError)?;
+                let closed_schema =
+                    ClosedDataType::from_resolve_data((**data_type).clone(), &closed_metadata)
+                        .expect("Could not create closed schema");
```

Comment:
In case this bubbles up when creating a closed entity type schema (unless there's something else which will distinguish it, in which case feel free to ignore). Also applies further below
```suggestion
                        .expect("Could not create closed data type schema");
```

Reply:
That code should not be here at all, it should not panic (`expect` errors), but propagate the error up. The error, however, contains the information we want. We mainly use expect for tests and a few times if we are sure that this will not happen or would lead to unexpected behavior when preconditions are not met (bug at the implementation). Fixed in 6cf493f8cb9a0265c89d23c082d052543215a3f7